### PR TITLE
add Windows support

### PR DESCRIPTION
**Pull request by matthiasdiener**
Build instructions were mostly taken from https://github.com/vim/vim-win32-installer/blob/master/appveyor.bat

Followup of:

- #1257 
- #865

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is ready for a first look @bdice @kozo2

**Comment by bdice** [comment]
@matthiasdiener Very nice! This appears to fix the issues with runtime files that I had in my previous attempt. Were you able to verify that the runtime files are working as expected when using the resulting `vim.exe`?

**Comment by matthiasdiener** [comment]
> @matthiasdiener Very nice! This appears to fix the issues with runtime files that I had in my previous attempt. Were you able to verify that the runtime files are working as expected when using the resulting `vim.exe`?

I haven't verified that vim works correctly on Windows yet, I was hoping one of you could try if it works ;-) For some reason, azure doesn't seem to save the build artifact though.

**Comment by matthiasdiener** [comment]
@conda-forge-admin, rerender

**Comment by kozo2** [comment]
> > @matthiasdiener Very nice! This appears to fix the issues with runtime files that I had in my previous attempt. Were you able to verify that the runtime files are working as expected when using the resulting `vim.exe`?
> 
> I haven't verified that vim works correctly on Windows yet, I was hoping one of you could try if it works ;-) For some reason, azure doesn't seem to save the build artifact though.

I tried to build this repo locally (in my Windows env) to try if it works but the build didn't run as below
```
(base) C:\Users\kozo2\Documents\GitHub>conda build vim-feedstock
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.16
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.16
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Skipped: vim from C:\Users\kozo2\Documents\GitHub\vim-feedstock\recipe defines build/skip for this configuration ({'c_compiler': 'vs2017', 'cxx_compiler': 'vs2017', 'target_platform': 'win-64', 'python': '3.9'}).
Renaming work directory 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work' to 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop'
INFO:conda_build.utils:Renaming work directory 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work' to 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop'
shutil.move(work)=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work, dest=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop)
INFO:conda_build.utils:shutil.move(work)=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work, dest=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop)
# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:


# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes
anaconda upload ^

anaconda_upload is not set.  Not uploading wheels: []

INFO :: The inputs making up the hashes for the built packages are as follows:
{}


####################################################################################
Resource usage summary:

Total time: 0:00:00.1
CPU usage: sys=0:00:00.0, user=0:00:00.0
Maximum memory usage observed: 0B
Total disk usage observed (not including envs): 0B
```
Do you have any ideas?

**Comment by matthiasdiener** [comment]
> > > @matthiasdiener Very nice! This appears to fix the issues with runtime files that I had in my previous attempt. Were you able to verify that the runtime files are working as expected when using the resulting `vim.exe`?
> > 
> > 
> > I haven't verified that vim works correctly on Windows yet, I was hoping one of you could try if it works ;-) For some reason, azure doesn't seem to save the build artifact though.
> 
> I tried to build this repo locally (in my Windows env) to try if it works but the build didn't run as below
> 
> ```
> (base) C:\Users\kozo2\Documents\GitHub>conda build vim-feedstock
> No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.16
> WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.16
> Adding in variants from internal_defaults
> INFO:conda_build.variants:Adding in variants from internal_defaults
> Skipped: vim from C:\Users\kozo2\Documents\GitHub\vim-feedstock\recipe defines build/skip for this configuration ({'c_compiler': 'vs2017', 'cxx_compiler': 'vs2017', 'target_platform': 'win-64', 'python': '3.9'}).
> Renaming work directory 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work' to 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop'
> INFO:conda_build.utils:Renaming work directory 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work' to 'C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop'
> shutil.move(work)=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work, dest=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop)
> INFO:conda_build.utils:shutil.move(work)=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work, dest=C:\Users\kozo2\mambaforge\conda-bld\vim_1632869536540\work_moved_vim-8.2.3427-py39h82bb817_2_win-64_main_build_loop)
> # Automatic uploading is disabled
> # If you want to upload package(s) to anaconda.org later, type:
> 
> 
> # To have conda build upload to anaconda.org automatically, use
> # $ conda config --set anaconda_upload yes
> anaconda upload ^
> 
> anaconda_upload is not set.  Not uploading wheels: []
> 
> INFO :: The inputs making up the hashes for the built packages are as follows:
> {}
> 
> 
> ####################################################################################
> Resource usage summary:
> 
> Total time: 0:00:00.1
> CPU usage: sys=0:00:00.0, user=0:00:00.0
> Maximum memory usage observed: 0B
> Total disk usage observed (not including envs): 0B
> ```
> 
> Do you have any ideas?

I think this might be happening since you need to use the branch of this PR, no the `master` branch of the repo.

In any case, the artifacts are fixed now; Could you please take a look if this works for you: 

https://artprodeus21.artifacts.visualstudio.com/A910fa339-c7c2-46e8-a579-7ea247548706/84710dde-1620-425b-80d0-4cf5baca359d/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbmRhLWZvcmdlL3Byb2plY3RJZC84NDcxMGRkZS0xNjIwLTQyNWItODBkMC00Y2Y1YmFjYTM1OWQvYnVpbGRJZC8zODM4MTQvYXJ0aWZhY3ROYW1lL2NvbmRhX2FydGlmYWN0c18zODM4MTRfd2luXzY0Xw2/content?format=zip

**Comment by kozo2** [comment]
@matthiasdiener Thank you.
I tried the artifacts, and installed it with:
```
(base) C:\Users\kozo2\Desktop\conda_artifacts_383814_win_64_\win-64>conda install vim-8.2.3427-h0e60522_3.tar.bz2
```
and it seems that it works like the following attached image.

image

image

By the way, `gvim.exe` did not exist, and the `version:` of the `vim.exe` is `-python3` instead of `+python3`.
Does this match your expectations?

**Comment by matthiasdiener** [comment]
> @matthiasdiener Thank you. I tried the artifacts, and installed it with:
> 
> ```
> (base) C:\Users\kozo2\Desktop\conda_artifacts_383814_win_64_\win-64>conda install vim-8.2.3427-h0e60522_3.tar.bz2
> ```
> 
> and it seems that it works like the following attached image.

Great, thank you for checking!

> By the way, `gvim.exe` did not exist, and the `version:` of the `vim.exe` is `-python3` instead of `+python3`. Does this match your expectations?

Yes, this matches my (current) expectations. gvim will probably continue to not be built, but I'll look again if I can enable Python support.

File: recipe/bld.bat [hunk 1]
```diff
@@ -0,0 +1,26 @@
+echo on
+
+:: Build
+
+cd %SRC_DIR%\\src
```
Comment (bdice) [inline recipe/bld.bat:5]: @matthiasdiener I saw tests are failing with an error including `Python.h`:
```
if_python.c(66): fatal error C1083: Cannot open include file: 'Python.h': No such file or directory
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\cl.exe"' : return code '0x2'
Stop.
```

Maybe this script needs an include directory to be set like this?
https://github.com/vim/vim-win32-installer/blob/0f1f66596ddb80e18707934b459ee05e0392640e/appveyor.bat#L209

I'm not sure what the correct environment variable is ([list of conda environment variables](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html?highlight=library_prefix#environment-variables-set-during-the-build-process)). Maybe try:
```batch
set INCLUDE=%LIBRARY_INC%;%INCLUDE%
```

Comment (matthiasdiener) [reply recipe/bld.bat:16]: Thanks, that fixed it!

**Comment by matthiasdiener** [comment]
Well, that seems to have worked. Could you please check any of the Windows artifacts here to see if they are running? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=385244&view=artifacts&pathAsName=false&type=publishedArtifacts @bdice @kozo2

**Comment by kozo2** [comment]
@matthiasdiener Now it seems that there is no problem.
`:version` has `+python3/dyn` and `:!python --version` prints `Python 3.9.7`

**Comment by bdice** [comment]
@matthiasdiener @kozo2 The package artifacts work great for me! I verified that the option `+python3/dyn` is set and `:verbose set pythonthreedll?` returns `pythonthreedll=python39.dll`. I do have `Library/bin` on my `PATH`, so it just worked immediately as `vim.exe` without specifying the full path. Thanks @matthiasdiener for your hard work on this! 🚀

**Review by bdice — APPROVED** [review]

**Comment by matthiasdiener** [comment]
Thank you for checking! As far as I can tell, LIBRARY_BIN is the right location for the files, so I'll go ahead and merge this PR. Please let me know if you notice issues with the windows package.

**Pull request merged**